Index: a/xCAT-server/lib/xcat/plugins/dhcp.pm
===================================================================
--- a/xCAT-server/lib/xcat/plugins/dhcp.pm	(revision 12077)
+++ b/xCAT-server/lib/xcat/plugins/dhcp.pm	(working copy)
@@ -53,27 +53,54 @@
 my $chainents;
 my $tftpdir = xCAT::TableUtils->getTftpDir();
 use Math::BigInt;
-my $dhcpconffile = $^O eq 'aix' ? '/etc/dhcpsd.cnf' : '/etc/dhcpd.conf'; 
+my ($dhcpconffile, $dhcp6conffile) = get_conffile();
+my ($dhcpleases, $dhcp6leases) = get_leasesfile();
+my $usingipv6;
 my %dynamicranges; #track dynamic ranges defined to see if a host that resolves is actually a dynamic address
 my %netcfgs;
 my $distro = xCAT::Utils->osver();
 
-# dhcp 4.x will use /etc/dhcp/dhcpd.conf as the config file
-my $dhcp6conffile;
-if ( $^O ne 'aix' and -d "/etc/dhcp" ) {
-    $dhcpconffile = '/etc/dhcp/dhcpd.conf';
-    $dhcp6conffile = '/etc/dhcp/dhcpd6.conf'; 
+sub get_conffile {
+    my $dhcp6conffile = xCAT::Utils->get_site_attribute('dhcp6conf');
+    my $dhcpconffile = xCAT::Utils->get_site_attribute('dhcpconf');
+    if ( ! $dhcpconffile )
+    {
+        $dhcpconffile="/etc/dhcpd.conf";
+        # dhcp 4.x will use /etc/dhcp/dhcpd.conf as the config file
+        if ( $^O ne 'aix' and -d "/etc/dhcp" ) {
+            $dhcpconffile = '/etc/dhcp/dhcpd.conf';
+        }
+        # is this ubuntu ?
+        if ( $distro =~ /ubuntu.*/ ){
+            $dhcpconffile="/etc/dhcp3/dhcpd.conf";
+        }
+    }
+    if ( ! $dhcp6conffile )
+    {
+        $dhcp6conffile="/etc/dhcpd6.conf";
+        # dhcp 4.x will use /etc/dhcp/dhcpd.conf as the config file
+        if ( $^O ne 'aix' and -d "/etc/dhcp" ) {
+            $dhcp6conffile = '/etc/dhcp/dhcpd6.conf';
+        }
+    }
+
+    return ("$dhcpconffile", "$dhcp6conffile");
 }
-my $usingipv6;
 
-# is this ubuntu ?
-if ( $distro =~ /ubuntu.*/ ){
-    if (-e '/etc/dhcp/') {
-        $dhcpconffile = '/etc/dhcp/dhcpd.conf';
+sub get_leasesfile {
+    my $dhcp6leases = xCAT::Utils->get_site_attribute('dhcp6leases');
+    my $dhcpleases = xCAT::Utils->get_site_attribute('dhcpleases');
+    # default to whatever dhcpd runs normally
+    if ( ! $dhcpleases )
+    {
+        $dhcpleases="";
     }
-    else {
-        $dhcpconffile = '/etc/dhcp3/dhcpd.conf';	
+    if ( ! $dhcp6leases )
+    {
+        $dhcp6leases="";
     }
+
+    return ("$dhcpleases", "$dhcp6leases");
 }
 
 sub check_uefi_support {
@@ -1106,6 +1158,17 @@
             }
 
             my $ifarg = "$dhcpd_key=\"";
+            if ( $dhcpver eq "dhcpd" ) {
+                if ( not $dhcpleases eq "" ) {
+                    $ifarg .= " -lf ".$dhcpleases;
+                }
+                $ifarg .= " -cf ".$dhcpconffile;
+            } elsif ( $dhcpver eq "dhcpd6" ) {
+                $ifarg .= " -cf ".$dhcp6conffile;
+                if ( not $dhcp6leases eq "" ) {
+                    $ifarg .= " -lf ".$dhcp6leases;
+                }
+            }
             foreach (keys %activenics) {
                 if (/!remote!/) { next; }
                 $ifarg .= " $_";
