--- a/xCAT-server/lib/xcat/plugins/dhcp.pm	(revision 12185)
+++ b/xCAT-server/lib/xcat/plugins/dhcp.pm	(working copy)
@@ -1600,6 +1600,12 @@
                    "    min-lease-time $dhcplease;\n",
                    "    default-lease-time $dhcplease;\n",
                    );
+
+    my $namesearchpath = xCAT::TableUtils->get_site_attribute('namesearchpath');
+    if ( ! $namesearchpath) {
+        $namesearchpath = "";
+    }
+
     #for now, just do address allocatios (phase 1)
     #phase 2 (by 2.6 presumably) will include the various things like DNS server and other options allowed by dhcpv6
     #gateway is *not* currently allowed to be DHCP designated, router advertises its own self indpendent of dhcp.  We'll just keep it that way
@@ -1611,7 +1629,7 @@
     #boot url/param (rfc 59070)
     my $netdomain = $netcfgs{$net}->{domain};
     unless ($netdomain) { $netdomain = $site_domain; }
-    push @netent, "    option domain-name \"".$netdomain."\";\n";
+    push @netent, "    option domain-name \"".$netdomain." ".$namesearchpath."\";\n";

     #  add domain-search if not sles10 or rh5
     my $osv = xCAT::Utils->osver();
@@ -1692,6 +1711,7 @@
 
         # if here, means we found the idx before which to insert
         my $nettab = xCAT::Table->new("networks");
+        my $namesearchpath = "";
         my $nameservers;
         my $ntpservers;
         my $logservers;
@@ -1860,7 +1881,12 @@
         }
         if ($nameservers)
         {
-            push @netent, "    option domain-name \"$domain\";\n";
+            $namesearchpath = $domain;
+            my $tmp = xCAT::TableUtils->get_site_attribute('namesearchpath');
+            if ($tmp) {
+                $namesearchpath .= " ".$tmp;
+            }
+            push @netent, "    option domain-name \"$namesearchpath\";\n";
             push @netent, "    option domain-name-servers  $nameservers;\n";
         }

