Index: a/xCAT-server/lib/xcat/plugins/ddns.pm
===================================================================
--- a/xCAT-server/lib/xcat/plugins/ddns.pm	(revision 12075)
+++ b/xCAT-server/lib/xcat/plugins/ddns.pm	(working copy)
@@ -411,6 +411,16 @@
         my @forwarders = split /[ ,]/,$site_entry;
         $ctx->{forwarders}=\@forwarders;
     }
+    $stab = $sitetab->getAttribs({key=>'named_listen_rules'},['value']);
+    if ($stab and $stab->{value}) {
+        my @named_listen_rules = split /[ ,]/,$stab->{value};
+        $ctx->{named_listen_rules}=\@named_listen_rules;
+    }
+    $stab = $sitetab->getAttribs({key=>'named_v6_listen_rules'},['value']);
+    if ($stab and $stab->{value}) {
+        my @named_listen_rules = split /[ ,]/,$stab->{value};
+        $ctx->{named_v6_listen_rules}=\@named_listen_rules;
+    }
 
     $ctx->{zonestotouch}->{$ctx->{domain}}=1;
 	foreach (@networks) {
 		 if ($_->{domain}) {
@@ -855,7 +865,20 @@
     }
     unless ($gotoptions) {
         push @newnamed,"options {\n","\tdirectory \"".$ctx->{zonesdir}."\";\n";
-	push @newnamed,"\t\t//listen-on-v6 { any; };\n";
+        if ($ctx->{named_v6_listen_rules}) {
+            push @newnamed, "\tlisten-on-v6 {\n";
+            foreach (@{$ctx->{named_v6_listen_rules}}) {
+				push @newnamed, "\t\t$_;\n";
+            }
+            push @newnamed, "\t};\n";
+        }
+        if ($ctx->{named_listen_rules}) {
+            push @newnamed, "\tlisten-on {\n";
+            foreach (@{$ctx->{named_listen_rules}}) {
+				push @newnamed, "\t\t$_;\n";
+            }
+            push @newnamed, "\t};\n";
+        }
         if ($ctx->{forwarders}) {
             push @newnamed,"\tforwarders {\n";
             foreach (@{$ctx->{forwarders}}) {
